src/pkg/oci/common.go

```diff
@@ -75,38 +75,34 @@ func WithInsecureSkipVerify(insecure bool) Modifier {
 	}
 }
 
-// WithTargetPlatform sets the target platform for the remote
-func WithTargetPlatform(platform *ocispec.Platform) Modifier {
-	return func(o *OrasRemote) {
-		o.targetPlatform = platform
-	}
-}
-
 // WithSkeletonArch sets the target architecture for the remote to skeleton
-func WithSkeletonArch() Modifier {
-	return WithTargetPlatform(&ocispec.Platform{
+func WithSkeletonArch() ocispec.Platform {
+	return ocispec.Platform{
 		OS:           MultiOS,
 		Architecture: SkeletonArch,
-	})
+	}
 }
 
 // WithArch sets the target architecture for the remote
-func WithArch(arch string) Modifier {
-	return WithTargetPlatform(&ocispec.Platform{
+func WithArch(arch string) ocispec.Platform {
+	return ocispec.Platform{
 		OS:           MultiOS,
 		Architecture: arch,
-	})
+	}
 }
 
 // NewOrasRemote returns an oras remote repository client and context for the given url.
 //
 // Registry auth is handled by the Docker CLI's credential store and checked before returning the client
-func NewOrasRemote(url string, mods ...Modifier) (*OrasRemote, error) {
+func NewOrasRemote(url string, platform ocispec.Platform, mods ...Modifier) (*OrasRemote, error) {
 	ref, err := registry.ParseReference(strings.TrimPrefix(url, helpers.OCIURLPrefix))
 	if err != nil {
 		return nil, fmt.Errorf("failed to parse OCI reference %q: %w", url, err)
 	}
 	o := &OrasRemote{}
+	if platform.OS != "" || platform.Architecture != "" || platform.OSVersion != "" || len(platform.OSFeatures) != 0 || platform.Variant != "" {
+		o.targetPlatform = &platform
+	}
```

Comment:
```suggestion
```
I would rather force the platform to be given

Reply:
checks removed, now just setting the platform